src/api/controllers/socialAuth.js

```diff
@@ -0,0 +1,113 @@
+import model from '../models';
+import generateToken from '../../helpers/tokens/generate.token';
+
+const { User } = model;
+
+/**
+ *
+ *
+ * @class socialLogin
+ */
+class socialLogin {
+  /**
+   * signup || login via facebook
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async facebookAuth(req, res) {
+    let existingUser = await User.findOne({
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!existingUser) {
+      existingUser = await User.create({
+        username: req.user.name.familyName,
+        profileImage: req.user.photos[0].value,
```

Comment:
@Ramadhan0 what is the format of this `req.user.photos[0].value`? Is it a URL linking to their profile photo? Please, sync with @alainburindi to know his implementation of profile pictures.

Reply:
Alain is going to use this profile URL from social media.
Alain proposed that:  " if a user register using social link ie. facebook, the profile picture will automatically be the one from facebook until he or she decided to change the profile picture"